internal/upgrader/upgrader_test.go

```diff
@@ -181,55 +181,84 @@ var _ = Describe("Upgrade", func() {
+						GUID:                   "fake-plan-2-guid",
+						Available:              true,
+						Name:                   "fake-plan-2-name",
+						MaintenanceInfoVersion: "1.5.7",
+					},
+				}, nil)
 
-			By("getting the service plans")
-			Expect(fakeCFClient.GetServicePlansCallCount()).To(Equal(1))
-			Expect(fakeCFClient.GetServicePlansArgsForCall(0)).To(Equal(fakeBrokerName))
+				fakeInstance1 = ccapi.ServiceInstance{
+					GUID:                              "fake-instance-guid-1",
+					Name:                              "fake-instance-name-1",
+					UpgradeAvailable:                  false, // no upgradable, skipped instance
+					MaintenanceInfoVersion:            "1.3.0",
+					ServicePlanGUID:                   "fake-plan-1-guid",
+					ServicePlanMaintenanceInfoVersion: "1.5.7",
+					ServicePlanDeactivated:            false,
+				}
 
-			By("getting the service instances")
-			Expect(fakeCFClient.GetServiceInstancesForServicePlansCallCount()).To(Equal(1))
-			Expect(fakeCFClient.GetServiceInstancesForServicePlansArgsForCall(0)).To(Equal([]ccapi.ServicePlan{fakePlan}))
+				fakeInstance2 = ccapi.ServiceInstance{
+					GUID:                              "fake-instance-guid-2",
+					Name:                              "fake-instance-name-2",
+					UpgradeAvailable:                  true,
+					MaintenanceInfoVersion:            "1.5.7",
+					ServicePlanGUID:                   "fake-plan-1-guid",
+					ServicePlanMaintenanceInfoVersion: "1.5.7",
+					ServicePlanDeactivated:            false,
+				}
 
-			By("not calling upgrade")
-			Expect(fakeCFClient.UpgradeServiceInstanceCallCount()).Should(Equal(0))
+				// Plan and instance with different version
+				notUpToDateInstance = ccapi.ServiceInstance{
+					GUID:                              "fake-instance-guid-3",
+					UpgradeAvailable:                  true,
+					MaintenanceInfoVersion:            "1.5.7",
+					ServicePlanGUID:                   "fake-plan-2-guid",
+					ServicePlanMaintenanceInfoVersion: "1.5.0",
+					ServicePlanDeactivated:            false,
+				}
 
-			By("printing the GUIDs")
-			Expect(result).To(ContainSubstring(fmt.Sprintf("discovering service instances for broker: %s", fakeBrokerName)))
-			Expect(result).To(ContainSubstring("the following service instances would be upgraded:"))
-			Expect(result).To(ContainSubstring(fmt.Sprintf(`Service Instance GUID: "%s"`, fakeInstance1.GUID)))
-			Expect(result).To(ContainSubstring(fmt.Sprintf(`Service Instance GUID: "%s"`, fakeInstance2.GUID)))
-		})
+				withPlanDeactivated = ccapi.ServiceInstance{
+					GUID:                              "fake-instance-guid-4",
+					Name:                              "fake-instance-name-4",
+					UpgradeAvailable:                  true,
+					MaintenanceInfoVersion:            "1.3.0",
+					ServicePlanGUID:                   "fake-plan-1-guid",
+					ServicePlanMaintenanceInfoVersion: "1.5.7",
+					ServicePlanDeactivated:            true, // plan deactivated, skipped instance
```

Comment:
question: In this scenario the plan was deactivated, but there is already an upgrade available. I don't see in the code where we would be skipping it, it sounds more like the upgrade would fail in the cloud controller side when attempted. Although I might be wrong.